docs/advanced/customization-guide.md

```diff
@@ -30,8 +30,11 @@ labeling:
 ## NodeFeatureRule custom resource
 
 `NodeFeatureRule` objects provide an easy way to create vendor or application
-specific labels. It uses a flexible rule-based mechanism for creating labels
-based on node feature.
+specific labels and taints. It uses a flexible rule-based mechanism for creating
+labels/taints based on node features. Tainting feature is disabled by default and
+to enable it set `--no-taint` flag to `false`. If the feature gate `--no-taint`
+is `true` (i.e. disabled), taints defined in the NodeFeatureRule CRD have no effect
+and will be ignored by the NFD controller.
```

Comment:
Maybe we should add a note/warning that the taints will possibly cause `nfd-worker` pods to be evicted from the nodes that get tainted. And note that this should be negated by customizing the nfd-worker daemonset with corresponding tolerations. Or, even creating a generic toleration that tolerates all possible taints:

```
  tolerations:
  - operator: Exists
```

We could even move the taint-specific content here into a separate *Taints* sub-section (under the same *NodeFeatureRule custom resource* section)

WDYT?

Also, we should describe the new `taints` field in the *Label rule format* section below.

Reply:
Good point about eviction, I actually ran into the same situation with Tilt while re-deploying the NFD. I will document them as per suggestion.

Reply:
Another thought: if one would add a generic toleration I suggested above (` tolerations: [operator: Exists]`) that will mean that nfd-worker would likely be scheduled on K8s control-plane node(s) too. And that in turn, might cause control plane node(s) to be tainted and nfd-master be evicted 🫤 So, if that kind of über-toleration would be used that should be set to nfd-master, too.

Reply:
I've updated documentation commit, basically took it into a separate sub-section and added more info there. As for the toleration discussion, I added a note that user will need to add corresponding toleration on the Pod/DaemonSet. Should we add a note about generic toleration or leave it?